XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandBounds.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.EventDatum.Input;
+using XRTK.Interfaces.InputSystem;
+using XRTK.Interfaces.InputSystem.Handlers;
+using XRTK.Interfaces.Providers.Controllers;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    /// <summary>
+    /// Utility behavior to access the axis aligned bounds of IMixedRealityHands.
+    /// </summary>
+    public class HandBounds : MonoBehaviour, IMixedRealitySourceStateHandler, IMixedRealityHandJointHandler
+    {
+        /// <summary>
+        /// Accessor for the bounds associated with a handedness.
+        /// </summary>
+        public Dictionary<Handedness, Bounds> Bounds { get; private set; } = new Dictionary<Handedness, Bounds>();
+
+        [SerializeField]
+        [Tooltip("Should a gizmo be drawn to represent the hand bounds.")]
+        private bool drawBoundsGizmo = false;
+
+        /// <summary>
+        /// Should a gizmo be drawn to represent the hand bounds.
+        /// </summary>
+        public bool DrawBoundsGizmo
+        {
+            get { return drawBoundsGizmo; }
+            set { drawBoundsGizmo = value; }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        protected IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    inputSystem = MixedRealityToolkit.GetService<IMixedRealityInputSystem>();
```

Comment:
Just replace this with `MixedRealityToolkit.InputSystem` everywhere. There's really no need to write this logic again.